extractor/extraction_containers.cpp

```diff
@@ -479,6 +484,8 @@ void ExtractionContainers::WriteEdges(std::ofstream& file_out_stream) const
             continue;
         }
 
+        // IMPORTANT!!!!
+        // Here, we depend on slicing to write out the base class (not NodeBasedEdgeWithOSM)
         file_out_stream.write((char*) &edge.result, sizeof(NodeBasedEdge));
         number_of_used_edges++;
```

Comment:
I changed this to create a copy of the base class, and *really* depend on slicing to get just the data we want.